subprojects/model-core/src/test/groovy/org/gradle/internal/reflect/validation/ValidationMessageCheckerTest.groovy

```diff
@@ -770,6 +770,72 @@ Please refer to https://docs.gradle.org/current/userguide/validation_problems.ht
 """
     }
 
+    @ValidationTestFor(
+        ValidationProblemId.UNKNOWN_IMPLEMENTATION
+    )
+    def "tests output of unknown implementation of nested property implemented by lambda"() {
+        when:
+        render implementationUnknown(true) {
+            nestedProperty('action')
+            implementedByLambda('LambdaAction')
+            includeLink()
+        }
+
+        then:
+        outputEquals """
+Property 'action' was implemented by the Java lambda 'LambdaAction\$\$Lambda\$<non-deterministic>'. Using Java lambdas is not supported, use an (anonymous) inner class instead.
+
+Reason: Gradle cannot track inputs when it doesn't know their implementation.
```

Comment:
```suggestion
Reason: Gradle cannot track inputs when it cannot uniquely identify their implementation.
```

Reply:
The reason is now dependent on the kind of unknown implementation, please take another look.